addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,333 @@
+  }
+
+  /**
+   * Populates the topics array with an entry for each cell associated with a forum being handled.
+   * In most cases that means all topic cells, but Search pages are special.
+   * Any topics gathered previously will be forgotten. (And should have already been when disabling.)
+   * Each topic is:
+   *    - topicId      int, used to compare topic "age" relative to other posts on the page
+   *    - topicCell    Node that will be highlighted if too old
+   *    - forum        String, usually the forum being listed. Comes from the row on a search page
+   *    - restoreCell  Only defined if the cell is modified.
+   *                   A clone of the initial state of the cell Node, to revert for disable addon
+   */
+  async function gatherTopics() {
+    topics = [];
+    let possibleTopicCells; // verified before forming a topic
+    if (isMobileSite) {
+      possibleTopicCells = document.querySelectorAll(".topic.list>li");
+    } else {
+      possibleTopicCells = document.querySelectorAll(".tcl");
+    }
+    // console.log("possibleTopicCells length: " + possibleTopicCells.length);
+    let theForum = forumOrSearchPageName; // usually, but overridden per cell if on a search page
+    // console.log("theForum: " + theForum);
+    for (let i = 0; i < possibleTopicCells.length; i++) {
+      if (forumOrSearchPageName.includes(searchResultsPageName)) {
+        // the cell to the right of the topic cell lists the forum, in search results
+        if (!possibleTopicCells[i].nextElementSibling) {
+          // synthetic "Page n" rows attached by Infinite Scrolling on search pages
+          // have no sibling, because they span all 4 columns
+          continue;
+        }
+        theForum = possibleTopicCells[i].nextElementSibling.innerText;
+      }
+      const theTopicId = await extractTopicIdFrom(possibleTopicCells[i]);
+      if (theTopicId != 0) {
+        // No restoreCell property at this point. Only added if the cell is modified
+        const topic = {
+          topicId: theTopicId,
+          topicCell: possibleTopicCells[i],
+          forum: theForum,
+        };
+        topics.push(topic);
+        // console.log("gatherTopics: " + topic.topicId + " in " + topic.forum);
+      }
+    }
+  }
+
+  /**
+   * How many topics since the earliest on the same page to be considered a "necropost"
+   */
+  function determineTopicInterval() {
+    if (!addon.settings.get("chooseCustomTopicInterval")) {
+      return defaultTopicInterval;
+    }
+    return addon.settings.get("customTopicInterval");
+  }
+
+  /**
+   * Restores topic cells to the initial state they had before any highlighting on this page.
+   * Clears any previously gathered topics.
+   * Does not conflict with Addon: Website dark mode and customizable colors.
+   */
+  async function removeAnyPriorHighlights() {
+    topics.forEach((topic) => {
+      if (topic.restoreCell) {
+        topic.topicCell.replaceWith(topic.restoreCell);
+      }
+    });
```

Comment:
Maybe implement colors as a style? We shouldn't waste memory on something that can be implemented as a userstyle

Reply:
I initially just added a style background property to highlight cells and removed all such properties to deactivate. When I added the feature to replace link text or append "Necropost?", I instead started capturing the state before my change in a "restoreCell" to revert all changes at once.

Is the memory used to store copies of a couple cells on the page really significant?

Reply:
It's not significant, but it could help multi-tasking